lib/PuppeteerSharp/ElementHandle.cs

```diff
@@ -1,26 +1,111 @@
 ﻿using System;
+using System.Collections.Generic;
+using System.Linq;
 using System.Threading.Tasks;
 
 namespace PuppeteerSharp
 {
     public class ElementHandle : JSHandle
     {
-        private Page _page;
+        private readonly Page _page;
 
-        public ElementHandle(ExecutionContext context, Session client, object args, Page page) :
-            base(context, client, args)
+        public ElementHandle(ExecutionContext context, Session client, object remoteObject, Page page) :
+            base(context, client, remoteObject)
         {
             _page = page;
         }
 
+        public override ElementHandle AsElement() => this;
+
+        public async Task ClickAsync(Dictionary<string, object> options = null)
+        {
+            var (x, y) = await GetVisibleCenterAsync();
+            await _page.Mouse.Click(x, y, options ?? new Dictionary<string, object>());
+        }
+
         internal Task TapAsync()
         {
             throw new NotImplementedException();
         }
 
+        internal async Task<ElementHandle> GetElementAsync(string selector)
+        {
+            var handle = await ExecutionContext.EvaluateFunctionHandleAsync("(element, selector) => element.querySelector(selector)", this, selector);
+
+            var element = handle.AsElement();
+            if (element != null)
+            {
+                return element;
+            }
+
+            await handle.Dispose();
+            return null;
+        }
+
         internal Task DisposeAsync()
         {
             throw new NotImplementedException();
         }
+
+        private async Task<(decimal x, decimal y)> GetVisibleCenterAsync()
```

Comment:
Though I'd use the `Get` prefix, in another library, I want to be as close as possible as `Puppeteer` code, let's go with `VisibleCenterAsync`. I know I know, async is not part of the original, but we need it :p